### PR TITLE
Overhaul test suite

### PR DESCRIPTION
**Pull request by 0xViola**
Refactored all tests to utilize the TestUtils.sol module which exports live auctions and bids for users to sign and for the settlement contract to asynchronously mint with.

TestUtils also provides the ability to regenerate auctions with a provided address parameter which is necessary in the case of a proxy. In the proxy structure's case, the settlementUUPS logic implementation is no longer serving the minter role as it does with a standalone contract, rather the ProxyDeoxys 1967 proxy is the one serving the minter role. Bids are regenerated simultaneously to update their target Pikapatible auctions.

TL;DR:
This PR makes the Settlement, SettlementUUPS, and ProxyDeoxys test files more compact + readable via code reuse, as well as allows the bids/auctions generated in TestUtils.sol to be used and regenerated on a whim by all tests across the entire repo.

### CODE REVIEW

**Review by ash-keccakum — APPROVED** [review]
🔥 LGTM! Nice!
